common/authen.js

```diff
@@ -68,6 +68,56 @@
 
             },
 
+            loginInANewWindow: function(cb) {
+                var serviceURL = (chaiseConfig.ermrestLocation ? chaiseConfig.ermrestLocation : $window.location.origin + "/ermrest");
+                var referrerId = (new Date().getTime());
+                var url = serviceURL + '/authn/preauth?referrer=' + UriUtils.fixedEncodeURIComponent(window.location.href.substring(0,window.location.href.indexOf('chaise')) + "chaise/login?referrerid=" + referrerId);
+                var config = {
+                    headers: {
+                        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+                        'Accept': 'application/json'
+                    }
+                };
+
+                window.addEventListener('message', function(args) {
+                    console.log("Inside message listener");
+                    if (args && args.data && (typeof args.data == 'string')) {
```

Comment:
I am unsure about identifying the identity as tabs usually are incapable of accessing the url's of other tabs. Whenever you want to post a message to a window, you need to have  a handle to it. Which in our case the window has it in `window.opener`. So we really won't need that information.